controllers/profile.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"net/http"
+
+	"google.golang.org/appengine/datastore"
+
+	"encoding/json"
+
+	"github.com/coduno/app/model"
+	"github.com/coduno/engine/util"
+	"github.com/gorilla/mux"
+	"golang.org/x/net/context"
+)
+
+func GetProfileByKey(ctx context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
+	if err := util.CheckMethod(r, "GET"); err != nil {
+		return http.StatusMethodNotAllowed, err
+	}
+
+	key, err := datastore.DecodeKey(mux.Vars(r)["key"])
+
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	var profile model.Profile
+	if err := datastore.Get(ctx, key, &profile); err != nil {
+		return http.StatusInternalServerError, err
+	}
+	json.NewEncoder(w).Encode(profile.Key(key))
+	return http.StatusOK, nil
+}
+
+func DeleteProfile(ctx context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
+	if err := util.CheckMethod(r, "DELETE"); err != nil {
+		return http.StatusMethodNotAllowed, err
+	}
+
+	key, err := datastore.DecodeKey(mux.Vars(r)["key"])
+
+	if err != nil {
+		return http.StatusInternalServerError, err
```

Comment:
This is an `http.StatusBadRequest`, because the key sent by the client cannot be decoded. This is probably because it arrived in a wrong format.